tests/manager/celery/tasks/test_generate_inventory_and_hold_reports.py

```diff
@@ -199,60 +202,75 @@ def test_job_run(
     assert "Inventory and Holds Reports" in kwargs["subject"]
     attachments: dict = kwargs["attachments"]
 
-    assert len(attachments) == 2
-    inventory_report_key = [x for x in attachments.keys() if "inventory" in x][0]
-    assert inventory_report_key
-    assert "test_library" in inventory_report_key
-    inventory_report_value = attachments[inventory_report_key]
-    assert inventory_report_value
-    inventory_report_csv = list(csv.DictReader(open(inventory_report_value)))
-
-    assert len(inventory_report_csv) == 1
-    for row in inventory_report_csv:
-        assert row["title"] == title
-        assert row["author"] == author
-        assert row["identifier"]
-        assert row["language"] == language
-        assert row["publisher"] == publisher
-        assert row["audience"] == "young adult"
-        assert row["genres"] == "genre_a,genre_z"
-        assert row["format"] == edition.BOOK_MEDIUM
-        assert row["data_source"] == data_source
-        assert row["collection_name"] == collection_name
-        assert float(row["days_remaining_on_license"]) == float(days_remaining)
-        assert row["shared_active_loan_count"] == "0"
-        assert row["library_active_loan_count"] == "0"
-        assert row["remaining_loans"] == str(checkouts_left)
-        assert row["allowed_concurrent_users"] == str(terms_concurrency)
-        assert expiration.strftime("%Y-%m-%d %H:%M:%S.%f") in row["license_expiration"]
-
-    holds_report_key = [x for x in attachments.keys() if "holds" in x][0]
-    assert holds_report_key
-    assert "test_library" in holds_report_key
-    holds_report_value = attachments[holds_report_key]
-    assert holds_report_value
-    holds_report_csv = list(csv.DictReader(open(holds_report_value)))
-    assert len(holds_report_csv) == 1
-
-    for row in holds_report_csv:
-        assert row["title"] == title
-        assert row["author"] == author
-        assert row["identifier"]
-        assert row["language"] == language
-        assert row["publisher"] == publisher
-        assert row["audience"] == "young adult"
-        assert row["genres"] == "genre_a,genre_z"
-        assert row["format"] == edition.BOOK_MEDIUM
-        assert row["data_source"] == data_source
-        assert row["collection_name"] == collection_name
-        assert int(row["shared_active_hold_count"]) == shared_patrons_in_hold_queue
-        assert int(row["library_active_hold_count"]) == 3
+    assert len(attachments) == 1
+    reports_zip = list(attachments.values())[0]
+    with zipfile.ZipFile(reports_zip, mode="r") as archive:
+        entry_list = archive.namelist()
+        assert len(entry_list) == 2
+        for entry in entry_list:
+            open_file = archive.open(entry)
```

Comment:
If you `open` without using a context manager, you have to call `close` or you will leak a file handle. Not a huge deal since its in tests, but it would be nice to avoid it. I think this can easily be reworked to use a context manager.